examples/web-wagmi/src/profiles/UseUpdateFollowPolicy.tsx

```diff
@@ -0,0 +1,213 @@
+import {
+  Amount,
+  ChargeFollowPolicy,
+  Erc20,
+  FollowPolicyType,
+  getFollowPolicyTypeFromProfileFieldsFragment,
+  NoFeeFollowPolicy,
+  ProfileFieldsFragment,
+  useCurrencies,
+  useUpdateFollowPolicy,
+} from '@lens-protocol/react';
+import { useState } from 'react';
+
+import { UnauthenticatedFallback } from '../components/UnauthenticatedFallback';
+import { WhenLoggedInWithProfile } from '../components/auth/auth';
+import { Loading } from '../components/loading/Loading';
+import { never } from '../utils';
+
+const followPolicyTypeToDescriptionMap: Record<FollowPolicyType, string> = {
+  [FollowPolicyType.ONLY_PROFILE_OWNERS]: 'Only holders of Lens profiles can follow',
+  [FollowPolicyType.ANYONE]: 'Anyone can follow',
+  [FollowPolicyType.NO_ONE]: 'No one can follow',
+  [FollowPolicyType.CHARGE]: 'Anyone can follow, but they must pay a fee',
+};
+
+function resolveFollowPolicy({
+  followPolicyType,
+  amount,
+  recipient,
+}: {
+  followPolicyType: FollowPolicyType;
+  amount?: Amount<Erc20>;
+  recipient?: string;
+}): ChargeFollowPolicy | NoFeeFollowPolicy {
+  if (followPolicyType === FollowPolicyType.CHARGE) {
+    return {
+      type: FollowPolicyType.CHARGE,
+      amount: amount ?? never(),
+      recipient: recipient ?? never(),
+    };
+  }
+
+  return {
+    type: followPolicyType,
+  };
+}
+
+type UpdateButtonTextProps = {
+  isTxPending: boolean;
+  currentFollowModule: FollowPolicyType | null;
+  followPolicyTypeToUpdate: FollowPolicyType | null;
+};
+
+function UpdateButtonText({
+  isTxPending,
+  currentFollowModule,
+  followPolicyTypeToUpdate,
+}: UpdateButtonTextProps) {
+  if (isTxPending) return <>Updating...</>;
+
+  if (currentFollowModule === followPolicyTypeToUpdate) {
+    return <>This is your current follow policy</>;
+  }
+
+  return <>Update</>;
+}
+
+type UpdateFollowPolicyProps = {
+  profile: ProfileFieldsFragment;
+};
+
+function UpdateFollowPolicy({ profile }: UpdateFollowPolicyProps) {
+  const currentFollowModule = getFollowPolicyTypeFromProfileFieldsFragment(profile);
+  const [followPolicyTypeToUpdate, setFollowPolicyTypeToUpdate] = useState<FollowPolicyType | null>(
+    currentFollowModule,
+  );
+
+  const { data: currencies, loading } = useCurrencies();
+  const { updateFollowPolicy, isPending, error } = useUpdateFollowPolicy();
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    if (!followPolicyTypeToUpdate || !currencies) return;
```

Comment:
nit: If it's possible to include `followPolicyTypeToUpdate` in the "FormData approach" we can also reduce the noise of currencies by either deferring the rendering of the form until currencies are fetched (requires an intermediate component sure but it will have more linear code.

Reply:
Thanks, I like this approach, amended the form to be in it's own component where all props are required - much more linear